lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -194,31 +198,59 @@ defaultRecordTypeOptions = Aeson.defaultOptions
 getStakePoolMetadata
     :: Trace IO RegistryLog
     -- ^ Logging object - use 'transformTrace' to convert to 'Text'.
+    -> FilePath
+    -- ^ Directory to cache the metadata registry.
     -> String
     -- ^ URL of a zip archive to fetch and extract.
     -> [PoolOwner]
     -- ^ List of stake pools to get metadata for.
     -> IO (Either FetchError [Maybe StakePoolMetadata])
-getStakePoolMetadata tr url poolOwners = fmap join $ tryJust fileExceptionHandler $
-    withSystemTempFile "stake-pool-metadata.zip" $ \zipFileName zipFile -> do
-        let tr' = contramap (fmap (RegistryLog url zipFileName)) tr
-        logTrace tr' MsgDownloadStarted
-        fetchStakePoolMetaZip url zipFile >>= \case
-            Right size -> do
-                logTrace tr' $ MsgDownloadComplete size
-                hClose zipFile
-                Right <$> getMetadataFromZip tr' zipFileName poolOwners
-            Left e -> pure $ Left e
+getStakePoolMetadata tr cacheDir url poolOwners = do
+    let zipFileName = cacheDir </> "registry.zip"
+    let tr' = contramap (fmap (RegistryLog url zipFileName)) tr
+    createDirectoryIfMissing False cacheDir
```

Comment:
**[question]**

Why not also create parent directory (i.e. ~~False~~ -> True) if they don't exist to easily allow creating nested directories? :thinking:

Reply:
Yeah, maybe. I was originally planning to bury this file in the database directory, which should exist. But actually the database is optional, so now it's now a temporary directory. So to resolve this question I now remove the createDirectoryIfMissing altogether.